### PR TITLE
Pull request Piotr Cieślik 202131

### PR DESCRIPTION
**Pull request by 202131**

### CODE REVIEW

File: src/main/java/edu/kis/powp/jobs2d/drivers/adapter/DrawPanelAdapter.java [hunk 1]
```diff
@@ -4,15 +4,17 @@
...
 import edu.kis.powp.jobs2d.Job2dDriver;
+import edu.kis.powp.jobs2d.features.DrawerFeature;
 
 /**
  * driver adapter to drawer with several bugs.
  */
-public class MyAdapter extends DrawPanelController implements Job2dDriver {
+public class DrawPanelAdapter extends DrawPanelController implements Job2dDriver {
 	private int startX = 0, startY = 0;
+	private DrawPanelController DPC;
```
Comment (LordIllidan) [inline src/main/java/edu/kis/powp/jobs2d/drivers/adapter/DrawPanelAdapter.java:14]: Jeśli zmienia Pan z adaptera klasy na adapter obiektu  powinniśmy usunąć dziedziczenie

**Review by LordIllidan — COMMENTED** [review]
Adapter odhaczam
Command wymaga dalszych prac

File: src/main/java/edu/kis/powp/jobs2d/drivers/adapter/DrawPanelAdapter.java [hunk 2]
```diff
@@ -4,15 +4,17 @@
...
 
-	public MyAdapter() {
-		super();
+	public DrawPanelAdapter() {
+	DPC = DrawerFeature.getDrawerController();
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/drivers/adapter/DrawPanelAdapter.java:17]: Constructor injection should be used (using a parameter). This is unnecessary direct coupling with DrawerFeature class.

File: src/main/java/edu/kis/powp/jobs2d/drivers/adapter/DrawPanelAdapter.java [hunk 3]
```diff
@@ -4,15 +4,17 @@
 import edu.kis.legacy.drawer.shape.ILine;
 import edu.kis.legacy.drawer.shape.LineFactory;
 import edu.kis.powp.jobs2d.Job2dDriver;
+import edu.kis.powp.jobs2d.features.DrawerFeature;
 
 /**
  * driver adapter to drawer with several bugs.
  */
-public class MyAdapter extends DrawPanelController implements Job2dDriver {
+public class DrawPanelAdapter extends DrawPanelController implements Job2dDriver {
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/drivers/adapter/DrawPanelAdapter.java:12]: The name is better but it does not indicate the purpose of the adapter.

File: .idea/codeStyles/codeStyleConfig.xml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+<component name="ProjectCodeStyleConfiguration">
```
Comment (kowallus) [inline .idea/codeStyles/codeStyleConfig.xml:1]: Commits polluted with project files

File: src/test/java/edu/kis/powp/jobs2d/events/SelectTestFigureOptionListener.java [hunk 5]
```diff
@@ -16,6 +17,14 @@ public SelectTestFigureOptionListener(DriverManager driverManager) {
 
 	@Override
 	public void actionPerformed(ActionEvent e) {
-		FiguresJoe.figureScript1(driverManager.getCurrentDriver());
+		DrawerFeature.getDrawerController().clearPanel();
+	    if(e.getActionCommand().equals("Figure Joe 1")){
```
Comment (kowallus) [inline src/test/java/edu/kis/powp/jobs2d/events/SelectTestFigureOptionListener.java:21]: Menu action should not depend on its caption. Enum or separate listeners would be better. At least you should consider constants.

**Review by kowallus — COMMENTED** [review]
Additional remarks are inlined in the code.
3.3 and UML diagram (3.2.4) is missing.

**Pull request closed without merging**